T4XJYT_LGI301/T4XJYT_LGI301/Core/API/ApiDataProvider.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace T4XJYT_LGI301.Core.API
+{
+    public class ApiDataProvider : IApiDataProvider
```

Comment:
A HTTPClient lass amit field-ként birtokol ez az osztály implementálja az IDisposable interfészt, így a birtokló osztálynak is kellene ezt implementálnia és using blokk-ba kellene használni az osztályt.
Referenciák: https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient?view=net-8.0
https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/using-objects